paasta_tools/deployd/common.py

```diff
@@ -62,3 +69,27 @@ def rate_limit_instances(instances, number_per_minute, watcher_name):
 def exponential_back_off(failures, factor, base, max_time):
     seconds = factor * base ** failures
     return seconds if seconds < max_time else max_time
+
+
+def get_service_instances_with_changed_id(marathon_client, instances, cluster):
```

Comment:
Looking at the oracle code I think it's actually exactly what we want here. Whereas the config watchers just want to know if the newly configured app doesn't exist, here we want to know if there are any differences between configured and running apps which is what the oracle code does. I'm going to switch it to use that code.